### PR TITLE
HttpServerOperations#isFormUrlencoded handles header value with charset

### LINKED ISSUE #1988: HttpServerRequest.receiveForm() can't handle Content-Type application/x-www-form-urlencoded;charset=UTF-8
`HttpServerRequest.receiveForm()` uses `HttpServerOperations.isFormUrlencoded()` to check if the content is `application/x-www-form-urlencoded` but uses an exact match eg: `HttpHeaderValues.APPLICATION_X_WWW_FORM_URLENCODED.contentEqualsIgnoreCase(contentType)` but i have a web page attaching charset to the content type (`;charset=UTF-8`) and it throws an exception instead:

```java.lang.IllegalStateException: Request is not POST or does not have Content-Type with value 'application/x-www-form-urlencoded' or 'multipart/form-data'```

for `multipart/form-data` nettys `HttpPostRequestDecoder.isMultipart(nettyRequest)` is used instead which only checks if the Content-Type starts with `multipart/form-data`

## Possible Solution

Make `HttpServerOperations.isFormUrlencoded()` use `startsWith` instead of `contentEqualsIgnoreCase` 

* Reactor version(s) used: 4.1.70
* Other relevant libraries versions (eg. `netty`, ...): netty 1.0.13
* JVM version (`java -version`): java 11
* OS and version (eg. `uname -a`): windows 11

### PR DESCRIPTION
**Pull request by violetagg**
Fixes #1988

### CODE REVIEW

File: reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerOperations.java [hunk 1]
```diff
@@ -295,7 +295,8 @@ public HttpServerResponse headers(HttpHeaders headers) {
 	@Override
 	public boolean isFormUrlencoded() {
 		String contentType = requestHeaders().get(HttpHeaderNames.CONTENT_TYPE);
-		return HttpHeaderValues.APPLICATION_X_WWW_FORM_URLENCODED.contentEqualsIgnoreCase(contentType);
+		return contentType != null &&
+				contentType.startsWith(HttpHeaderValues.APPLICATION_X_WWW_FORM_URLENCODED.toString());
```
Comment (simonbasle) [inline reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerOperations.java:299]: I would suggest avoiding `startsWith` and instead maybe introducing some sort of headers sanitization or comparison util which removes/ignores directives... as you said offband, the `;` can be separated by whitespace so it is not a simple matter of checking the n-th character, unfortunately :(

File: reactor-netty-http/src/test/java/reactor/netty/http/server/HttpServerTests.java [hunk 2]
```diff
@@ -2717,4 +2717,40 @@ void testMaxKeepAliveRequestsBadValues(int maxKeepAliveRequests) {
...
+		doTestIsFormUrlencoded("application/x-www-form-urlencoded;charset=UTF-8", true);
+	}
+
+	@Test
+	void testIsFormUrlencodedWithoutCharset() {
+		doTestIsFormUrlencoded("application/x-www-form-urlencoded", true);
+	}
+
+	@Test
+	void testIsNotFormUrlencoded() {
```
Comment (simonbasle) [inline reactor-netty-http/src/test/java/reactor/netty/http/server/HttpServerTests.java:2732]: maybe add a test that `application/x-www-form-urlencoded-bad` is not `isFormUrlencoded`?

**Comment by simonbasle** [comment]
probably minor but we need to avoid false positives on top of avoiding the false negative of the original issue :)

**Review by simonbasle — CHANGES_REQUESTED** [review]
probably minor but we need to avoid false positives on top of the original false negative :)

File: reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerOperations.java (hunk 1 above)
Comment (violetagg) [reply reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerOperations.java:299]: good catch, let me fix that

Comment (violetagg) [reply reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerOperations.java]: please check now

**Review by simonbasle — APPROVED** [review]
Nice thinking of using the getMimetype util!

**Comment by violetagg** [comment]
@simonbasle Thanks for the review!

**Pull request merged**